components/DetailsCard/index.js

```diff
@@ -104,6 +104,7 @@ export default function DetailsCard() {
   return (
     <StyledDetailsCard>
       <h1>{activities.name}</h1>
+      <StyledJoinedmark>{joined.joined && <p>XX</p>}</StyledJoinedmark>
```

Comment:
This seems more complicated than it needs to be. 
Why don't you make a useState that keeps a Boolean

`const [isJoined, setIsJoined] = useState(false)`